fabrictestbed_extensions/fablib/fablib.py

```diff
@@ -2218,3 +2219,124 @@ def create_show_tableXXX(data, fields=None):
             for field in fields:
                 table.append([field, data[field]])
         return table
+
+    @staticmethod
+    def __can_allocate_node_in_worker(
+        worker: FimNode, node: Node, allocated: dict
+    ) -> Tuple[bool, str]:
+        """
+        Check if a node can be provisioned on a worker node on a site w.r.t available resources on that site
```

Comment:
This sounds like a server-side decision rather than client-side. Why are we doing this on client side?  Is there an equivalent logic on server side also?

Reply:
CF already checks/validates this. This enables early failure checks at the client side.